util/progress/progresswriter/multiwriter.go

```diff
@@ -0,0 +1,106 @@
+package progresswriter
+
+import (
+	"context"
+	"strings"
+	"sync"
+
+	"github.com/moby/buildkit/client"
+	"golang.org/x/sync/errgroup"
+)
+
+type MultiWriter struct {
```

Comment:
These new progress handling functions are copied from buildx with slight modification. Buildx versions will be removed on next vendor.